Source/JavaScriptCore/parser/Lexer.cpp

```diff
@@ -2501,6 +2501,21 @@ JSTokenType Lexer<T>::lexWithoutClearingLineTerminator(JSToken* tokenRecord, Opt
             U16_GET(m_code, 0, 0, m_codeEnd - m_code, codePoint);
             ASSERT(isIdentStart(codePoint));
         }
+
+        // We observe one character identifier very frequently because real world web pages are shipping minified JavaScript.
+        // This path handles it in a fast path.
+        // Keep in mind that isSingleCharacterIdentPart returns true for lead and trail surrogate. So this handles correctly
+        // if the first character is surrogate (in this case, this fast path will not be used since the second character is trail surrogate).
+        if (!isSingleCharacterIdentPart(peek(1))) {
+            const auto character = m_current;
+            shift();
+            if (lexerFlags.contains(LexerFlags::DontBuildKeywords))
+                tokenData->ident = nullptr;
+            else
+                tokenData->ident = makeIdentifier(std::span { &character, 1 });
```

Comment:
Don't we have single characters in a pool somewhere already?

Reply:
note that i'm not aware of any minifiers that output non-ascii identifiers

Reply:
It's probably only letters, in my experience anyway. It's probably also not uncommon for them to be 1 or 2 characters. Maybe we should specialize for exactly that?

Reply:
For now, I will just land this as is since Identifier needs to be a pointer.

Reply:
With the fix of course :P